intermediate_notebooks/examples/custreamz/parsing_haproxy_logs.ipynb

```diff
@@ -0,0 +1,427 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Parsing HAProxy Logs\n",
+    "\n",
+    "This notebook demonstrates real world log parsing examples. For this notebook we will use the popular HAProxy\n",
+    "format.\n",
+    "\n",
+    "This notebook will cover\n",
+    "+ Streaming log entry event level logs from Kafka using Streamz\n",
+    " - Those interested in Kafka can refer to: https://kafka.apache.org/quickstart to start a local Kafka cluster.\n",
+    "+ Counting log word occurances - (Word Count)\n",
+    "+ Calculating average backend response time\n",
+    "\n",
+    "Ok so what do these HAProxy logs look like? Well here is an example.\n",
+    "\n",
+    "```{\"logline\": \"[haproxy@10.0.0.1] <134>May 29 19:08:36 haproxy[113498]: 45.26.605.15:38738 [29/May/2019:19:08:36.691] HTTPS:443~ HTTP_ProvisionManagers/mp3 4/5/0/1/1 200 6182 - - --NI 3/3/0/0/0 0/0 {|} \"GET /v2/serverinfo HTTP/1.1\"}```\n",
+    "\n",
+    "It is unlikely you have a Kafka topic with these log messages in it already so lets generate some sample messages for you. First we need to install a few required dependencies and define some global configurations however.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Collecting package metadata (repodata.json): done\n",
+      "Solving environment: done\n",
+      "\n",
+      "# All requested packages already installed.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "!conda install -c conda-forge -c anaconda -y streamz python-confluent-kafka networkx graphviz"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Global Configurations"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import confluent_kafka as ck\n",
+    "from confluent_kafka.admin import AdminClient, NewTopic\n",
+    "\n",
+    "num_messages_to_produce = 500000\n",
+    "\n",
+    "kafka_brokers = ['localhost:9092'] # This is a list of your Kafka brokers and ports.\n",
+    "topic = 'haproxy-logs'\n",
+    "\n",
+    "kafka_conf = {'bootstrap.servers': kafka_brokers[0], 'group.id': 'custreamz', 'session.timeout.ms': 60000}  # Kafka configuration parameters. Any additional Kafka configurations can be placed here ...\n",
+    "\n",
+    "producer = ck.Producer(kafka_conf)  # Kafka producer\n",
+    "\n",
+    "# !!! - UNCOMMENT THE BELOW IF YOU WOULD LIKE TO HAVE THIS SCRIPT CREATE A KAFKA TOPIC FOR YOU !!!\n",
```

Comment:
It feels like this should be uncommented by default. The rest of the kafka configuration above assumes that the Kafka cluster being used is from either the Kafka quickstart, or from our Docker image, so the risk of the user accidentally creating spurious topics/messages on a "real" cluster seems low.